swagger_parser/lib/src/generator/templates/dart_enum_dto_template.dart

```diff
@@ -11,13 +11,9 @@ String dartEnumDtoTemplate(
   return '''
 import '${freezed ? 'package:freezed_annotation/freezed_annotation.dart' : 'package:json_annotation/json_annotation.dart'}';
 
+@JsonEnum()
 enum $className {
-  ${dataClass.items.map((e) => _valuePrefixForEnumItems(dataClass.type, e)).join(',\n  ')};
-
-  const $className();
-
-  factory $className.fromJson(Map<String, dynamic> json) =>
-      \$enumDecode(_\$${className}EnumMap, json);
+${dataClass.items.map((e) => _valuePrefixForEnumValueItems(dataClass.type, e)).join(',\n')};
```

Comment:
Why you not refactor valuePrefixForEnumItems and use valuePrefixForEnumValueItems? Enum type must be bool or int.